### PR TITLE
node part

### PR DESCRIPTION
**Pull request by Idrees-Mo**
with node

### CODE REVIEW

**Review by jackatomix — CHANGES_REQUESTED** [review]
@Idrees-Mo Thanks for submitting this.

In addition to a review of some of my other comments, could you also please consider the following change of scope:

Our FBaaS (FizzBuzz as a Service) product is exceedingly popular, and we're looking to add additional functionality. We would like the ability for users to optionally input their own Fizz and Buzz brackets if they'd like.

E.g. The current application has 2 brackets: `3 => Fizz` and `5 => Buzz`. But If I wanted to, I should be able to remove/update the default brackets and even add new ones like `10 => Foo`.

Could you please let me know how you'd tackle these updates, and what considerations you'd have for the security and performance of the API?

**Comment by Idrees-Mo** [comment]
> @Idrees-Mo Thanks for submitting this.
> 
> In addition to a review of some of my other comments, could you also please consider the following change of scope:
> 
> Our FBaaS (FizzBuzz as a Service) product is exceedingly popular, and we're looking to add additional functionality. We would like the ability for users to optionally input their own Fizz and Buzz brackets if they'd like.
> 
> E.g. The current application has 2 brackets: `3 => Fizz` and `5 => Buzz`. But If I wanted to, I should be able to remove/update the default brackets and even add new ones like `10 => Foo`.
> 
> Could you please let me know how you'd tackle these updates, and what considerations you'd have for the security and performance of the API?

@jackatomix - 
In that case, I think we would need to make it dynamic instead of a hardcoded one, like getting the options from users and doing something like below. 

const n = 50
const condition = {
  3: 'fizz',
  5: 'buzz',
  10: 'foo'
}

function fizzBuzz(n, condition){
  const fizzOrNumber = i => {
    const text = Object.keys(condition).reduce((acc, item) => {
      return i % parseInt(item) == 0 ? acc+condition[item] : acc;
    }, '');
     return text || i; 
  };

  let results = [];
  for(let i=1; i<n; i+=1){
    const item = Object.keys(condition).includes(`${i}`) ?  condition[i] : fizzOrNumber(i);
    results.push(item);
  }

  return results.join(',');
}

console.log(fizzBuzz(n, condition));

And for security, we may have a database for the app where not only each user would have an account but also be able to store their FBaaS. :)

File: 3-react-node-microservice/public/src/fizzbuzz/Fizzbuzz.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    height: 2rem;
+    font-size: large;
+  }
+  button {
+    background-color: #288cfb;
+    color: #fff;
+    padding: 0.2rem 1.8rem;
+    border-radius: 10px;
+    font-weight: bold;
+    border: 1.8px solid gray;
```
[the remark this answers is not in the document]
Comment (Idrees-Mo) [reply 3-react-node-microservice/public/src/fizzbuzz/Fizzbuzz.js:19]: You are right , I think I was just using rem everywhere so :)

File: 3-react-node-microservice/public/src/fizzbuzz/Fizzbuzz.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    font-size: 20px;
+    &:hover {
+      cursor: pointer;
+    }
+    &:focus {
+      outline: none;
+    }
```
[the remark this answers is not in the document]
Comment (Idrees-Mo) [reply 3-react-node-microservice/public/src/fizzbuzz/Fizzbuzz.js:26]: Thanks for the comment, I would def keep it in mind for a real project. :)

File: 3-react-node-microservice/public/src/fizzbuzz/Fizzbuzz.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    setEnd("");
+  };
+
+  return (
+    <div>
+      <Container onSubmit={onsumit}>
+        <Field>
+          <label>Start</label>
+          <input
+            type="text"
```
[the remark this answers is not in the document]
Comment (Idrees-Mo) [reply 3-react-node-microservice/public/src/fizzbuzz/Fizzbuzz.js:77]: Yes, that's another good tip thank for that :)

File: 3-react-node-microservice/public/src/fizzbuzz/Fizzbuzz.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            name="start"
+            value={start}
+            onChange={(e) => setStart(e.target.value)}
+          />
+        </Field>
+        <Field>
+          <label htmlFor="">End</label>
+          <input
```
[the remark this answers is not in the document]
Comment (Idrees-Mo) [reply 3-react-node-microservice/public/src/fizzbuzz/Fizzbuzz.js:85]: silly me :)

File: 3-react-node-microservice/public/src/fizzbuzz/Fizzbuzz.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  const onsumit = (e) => {
+    e.preventDefault();
+
+    if (isNaN(start) | isNaN(end)) {
+      return alert("number only");
+    }
+    const res = getData({ start, end });
+    setStart("");
+    setEnd("");
+  };
```
[the remark this answers is not in the document]
Comment (Idrees-Mo) [reply 3-react-node-microservice/public/src/fizzbuzz/Fizzbuzz.js:69]: In a real one, I would have a  state for any possible error and if there is one, show it in the UI for a better user experience. :)

**Pull request closed without merging**